core/io/resource_loader.cpp

```diff
@@ -691,7 +691,7 @@ Ref<Resource> ResourceLoader::_load_complete_inner(LoadToken &p_load_token, Erro
 			Error wtp_task_err = FAILED;
 			if (loader_is_wtp) {
 				// Loading thread is in the worker pool.
-				thread_load_mutex.unlock();
+				p_thread_load_lock.unlock();
```

Comment:
This helps with safety as this would have the lock be locked *and* the mutex itself unlocked, this ensures the mutex isn't operated on while the lock is existing

Reply:
Do changes here account for this: https://github.com/godotengine/godot/pull/93739#discussion_r1659827245?

Reply:
The unlock is equivalent so there's no difference, and see below